code/controllers/subsystem/non_firing/SSmaprotate.dm

```diff
@@ -73,4 +74,48 @@ SUBSYSTEM_DEF(maprotate)
 	else
 		log_startup_progress("There is no special rotation defined for this day")
 
-
+/datum/controller/subsystem/maprotate/proc/decide_next_map()
+	var/list/potential_maps = list()
+	for(var/x in subtypesof(/datum/map))
+		var/datum/map/M = x
+		if(!initial(M.voteable))
+			continue
+			// And of course, if the current map is the same
+		if(istype(SSmapping.map_datum, M))
+			continue
+			// Or the map from last round
+		if(istype(SSmapping.last_map, M))
+			continue
+		potential_maps[M] = 1
+	// We now have 3 maps. We then pick your highest priority map in the list. Does this mean votes 4 and 5 don't matter? Yeah, with this current system only your top 3 votes will ever be used. 4 and 5 are good info to know however!
+	for(var/mob/player in GLOB.player_list)
+		if(player.client)
+			var/placed_vote = FALSE
+			for(var/this_map as anything in player.client.prefs.map_vote_pref_json)
+				for(var/datum/god_I_hate as anything in potential_maps)
+					if("[god_I_hate]" == "[this_map]")
+						placed_vote = TRUE
+						potential_maps[god_I_hate]++ // We give it an assigned value that increases
+						break // We found the right map
+					continue
+				if(placed_vote)
+					break
+	var/list/returned_text = list()
+	returned_text += "Map Preference Vote Results:"
+	for(var/maps in potential_maps)
+		var/votes = potential_maps[maps]
+		var/percentage_text = ""
```

Comment:
Make this the default, it can be overridden if there is more than 1 vote
```suggestion
		var/percentage_text = "    0%"
```

Reply:
done